tests/metrics/test_stratification.py

```diff
@@ -1,316 +1,175 @@
-        return len(x) / len(sg)
-
-    assert set(sg.columns) == {
-        "age",
-        "sex",
-        "year",
-        "favorite_things",
-        "favorite_color",
-        "month",
-    }
-
-    # Age has different sized bins, so statistical tests are harder.
-    assert set(sg.age) <= set(rs.age_bins.age_group_name)
-
-    assert set(sg.sex) == {"Male", "Female"}
-    assert np.allclose(sg.groupby("sex").apply(proportion), 1 / 2, rtol=0.1)
-
-    years = set(sg.year)
-    assert len(years) == 1
-    assert list(years)[0] == str(stratification_time.year)
-
-    assert set(sg.favorite_things) == FAVORITE_THINGS
-    assert np.allclose(
-        sg.groupby("favorite_things").apply(proportion),
-        1 / len(FAVORITE_THINGS),
-        rtol=0.25,
+    builder.results.register_stratification.assert_any_call(
+        "event_year",
+        years_list,
+        rs.map_year,
+        is_vectorized=True,
+        requires_columns=["event_time"],
     )
-
-    assert set(sg.favorite_color) == set(FavoriteColor.OPTIONS)
-    assert np.allclose(
-        sg.groupby("favorite_color").apply(proportion),
-        1 / len(FavoriteColor.OPTIONS),
-        rtol=0.1,
+    builder.results.register_stratification.assert_any_call(
+        "entrance_year",
+        years_list,
+        rs.map_year,
+        is_vectorized=True,
+        requires_columns=["entrance_time"],
     )
+    builder.results.register_stratification.assert_any_call(
+        "exit_year",
+        years_list,
+        rs.map_year,
+        is_vectorized=True,
+        requires_columns=["exit_time"],
+    )
+    assert builder.results.register_stratification.call_count == 5
 
-    months = set(sg.month)
-    assert len(months) == 1
-    assert list(months)[0] == str(stratification_time.month)
-
-
-def test_group(stratifier_and_sim, default_levels, include_and_exclude):
-    rs, sim = stratifier_and_sim
-    sim.step()
-
-    include, exclude = include_and_exclude
-    expected_levels = (set(default_levels) | set(include)) - set(exclude)
-
-    pop = sim.get_population()
-    groups = list(rs.group(pop.index, include, exclude))
-
-    _check_groups(rs, expected_levels, groups, pop)
-
-
-def test_group_empty_index(stratifier_and_sim, default_levels, include_and_exclude):
-    rs, sim = stratifier_and_sim
-    sim.step()
-
-    include, exclude = include_and_exclude
-    expected_levels = (set(default_levels) | set(include)) - set(exclude)
-
-    pop = sim.get_population()
-    pop = pop[pop.age > 200]
-    groups = list(rs.group(pop.index, include, exclude))
 
-    _check_groups(rs, expected_levels, groups, pop)
+def test_results_stratifier_map_age_groups():
+    """Test that ages of the population are mapped to intervals as expected."""
+    pop = pd.DataFrame(FAKE_POP_AGE_DICT)
+    rs = ResultsStratifier()
+    rs.age_bins = pd.DataFrame(AGE_BINS_EXPECTED_DICT)
+    mapped_pop = rs.map_age_groups(pop)
+    assert mapped_pop.to_list() == FAKE_POP_EXPECTED_AGE_BINS_LIST
```

Comment:
Why are you converting the Series output here to a list and then comparing it to a list rather than have `FAKE_POP_EXPECTED_AGE_BINS_LIST` be a Series in the first place? Seems like an unnecessary step.

Reply:
I found it easier to define the expected values as dicts or lists.

Reply:
I've been able to change this.